src/main/java/org/sonarsource/scanner/jenkins/pipeline/WaitForQualityGateStep.java

```diff
@@ -0,0 +1,228 @@
+ *
+ * Jenkins Plugin for SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.scanner.jenkins.pipeline;
+
+import com.google.common.collect.ImmutableSet;
+import hudson.Extension;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.sonar.SonarInstallation;
+import hudson.plugins.sonar.action.SonarAnalysisAction;
+import hudson.plugins.sonar.client.HttpClient;
+import hudson.plugins.sonar.client.WsClient;
+import hudson.plugins.sonar.client.WsClient.ProjectQualityGate;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.jenkinsci.plugins.workflow.support.actions.PauseAction;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+public class WaitForQualityGateStep extends Step implements Serializable {
+
+  private static final Logger LOGGER = Logger.getLogger(WaitForQualityGateStep.class.getName());
+
+  public static class QGStatus implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final String status;
+
+    public QGStatus(String status) {
+      this.status = status;
+    }
+
+    @Whitelisted
+    public String getStatus() {
+      return status;
+    }
+
+  }
+
+  private String taskId;
+  private String installationName;
+
+  @DataBoundConstructor
+  public WaitForQualityGateStep() {
+    super();
+  }
+
+  @DataBoundSetter
+  public void setTaskId(@Nullable String taskId) {
+    this.taskId = taskId;
+  }
+
+  @DataBoundSetter
+  public void setInstallationName(String installationName) {
+    this.installationName = installationName;
+  }
+
+  public String getTaskId() {
+    return taskId;
+  }
+
+  public String getInstallationName() {
+    return installationName;
+  }
+
+  @Override
+  public StepExecution start(StepContext context) throws Exception {
+    return new Execution(this, context);
+  }
+
+  private static class Execution extends StepExecution implements SonarQubeWebHook.Listener {
+
+    private static final long serialVersionUID = 1L;
```

Comment:
Can you point me to an example? I quickly tried but I didn't find a way to create a pipeline job in my tests that was using my step (like if the extension was never loaded).